### PR TITLE
Allow the rock properties to be zero initialized without a deck.

### PR DESCRIPTION
**Pull request by blattms**
This is needed in parallel runs where a the rock properties will not
be read from the deck but be communicated from a master process. Nevertheless
we need to be able to initialize the data structures with the correct
container size. In addition we need to be able to change the container values
from opm-autodiff's BlackoilPropsDataHandle.

### CODE REVIEW

File: opm/core/props/rock/RockFromDeck.cpp [hunk 1]
```diff
@@ -72,6 +72,13 @@ namespace Opm
     {
     }
 
+    RockFromDeck::RockFromDeck(std::size_t number_of_cells)
+        : porosity_(number_of_cells, 0),
+          permeability_(number_of_cells*9, 0.0),
```
Comment (atgeirr) [inline opm/core/props/rock/RockFromDeck.cpp]: A comment explaining the 9 (room for full tensor in 3D represented as a matrix) would be good.

File: opm/core/props/rock/RockFromDeck.hpp [hunk 2]
```diff
@@ -31,10 +31,16 @@ namespace Opm
...
+        // access to porosity and permeability
+        friend class BlackoilPropsDataHandle;
+
     public:
         /// Default constructor.
         RockFromDeck();
-
+        /// Creates rock properties with zero porosity and permeability
+        /// \param number_of_cells The number of cells
+        RockFromDeck(std::size_t number_of_cells);
```
Comment (atgeirr) [inline opm/core/props/rock/RockFromDeck.hpp]: Should be explicit.

File: opm/core/props/rock/RockFromDeck.cpp (hunk 1 above)
Comment (blattms) [inline opm/core/props/rock/RockFromDeck.cpp]: Comment added. Pease note that 9 is used throught the file at various other places.

File: opm/core/props/rock/RockFromDeck.hpp (hunk 2 above)
Comment (blattms) [inline opm/core/props/rock/RockFromDeck.hpp]: I always miss this one. Thanks for the hint. Change applied

**Comment by atgeirr** [comment]
Thanks! Merging.

**Pull request merged**